### ISSUE TITLE
How are the masks calculated for length of stay, decomposition and in-hospital mortality?

### ISSUE THREAD

**Issue by salmanazarr** [YerevaNN/mimic3-benchmarks#62]
Apologies if this is a very simple question but I am confused about how & why masks are being computed for Length of Stay, Decomposition and In-hospital Mortality values returned by the Multi-Task reader.

**Comment by Harhro94** [comment]
When we are doing multitasking, for each input we have multiple outputs (mortality, remaining length of stay at each hour, decompensation at each hour and phenotypes). There are stays for which some of the outputs are not present. For example, when the stay is shorter than 2 days, the mortality label will not be provided. The masks are introduced to help us to keep everything in the same format. Moreover, later in the neural networks we will multiply the loss function of each output with the mask of that output. Whenever the mask is 0, that loss function will not be added into our total loss function.

**Comment by activatedgeek** [comment]
Oh I see that makes sense. Thank you so much for all the work on this dataset!

**Comment by activatedgeek** [comment]
Hey @Harhro94, I'm sorry for reopening the issue, but I observed in the Multi Task dataset, there are sequences which contain information of only < 48 hours. I tried looking for an explanation in the paper but could not. Sorry if I missed it. Do you mind telling how that case is handled?

**Comment by Harhro94** [comment]
That is possible, in that case the multitask network will try to predict length of stay, decompensation and phenotyping, but not in-hospital mortality. In general, for each sample we predict all labels that are present for that sample.

**Comment by Harhro94** [comment]
@salmanazarr, that is not a bug. Probably you are running the scripts on a small subset of the dataset, where there is no positive cases for some phenotype. If you run the script on the whole dataset, you will not get that error.
Another option is to catch that ValueError and return ROC AUC=nan.

**Comment by nftd** [comment]
when there is no `1` in the label, then `average=None` will not work.

`y_true = np.array([[0,0,1,1], [0,0,0,0]])
y_scores = np.array([[0.1, 0.4, 0.35, 0.6], [0.2, 0.3, 0.2, 0.9]])
roc_auc_score(y_true, y_scores, average=None)`
